driver/src/solution_submission/mod.rs

```diff
@@ -92,8 +103,15 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
         batch_index: U256,
         solution: Solution,
     ) -> Result<U256, SolutionSubmissionError> {
+        // NOTE: Compare with `>=` as the exchange's current batch index is the
+        //   one accepting orders and does not yet accept solutions.
+        while batch_index.as_u32() >= self.contract.get_current_auction_index()? {
+            debug!("Solved batch is not yet accepting solutions, waiting for next batch.");
+            thread::sleep(POLL_TIMEOUT);
```

Comment:
😢 Yet another thread sleep in the middle of the code.

Reply:
I know :sob: I couldn't think of a nicer way without making a larger change.

Reply:
@fleupold I just had an idea. We can make `"Solutions are no longer accepted for this batch"` a benign error if we are in the previous block and have the scheduler re-run the driver.

Reply:
The downside here is that it will re-fetch the orderbook and look for a solution again, but I don't think this should be a big issue.

Reply:
That's a nice idea. We briefly discussed it over slack. It would require fetching the last block before verifying and then return a benign retryable error.

I'm somewhat concerned about running the solver again unnecessarily, but if we run into this it's already a sign that we were finishing too early (however this might become the norm if we start solving before the batch is finalized). We could also introduce a caching layer for the price finder that can quickly return the same result if the input files are the same.

Reply:
> however this might become the norm if we start solving before the batch is finalized

I imagine at that point, the scheduler would be responsible for starting the solution submitter.

> We could also introduce a caching layer for the price finder that can quickly return the same result if the input files are the same.

That would be useful, especially when facing retryable errors.

Reply:
Overall I think the other approach might be a bit more work to implement, so I'd be in favor of merging it as is to reduce the amount of false warnings we will have next week and then look into improving this.

Reply:
Created #707 to capture this work.